manifests/contact.pp

```diff
@@ -0,0 +1,52 @@
+# = Define: sensu::routing
+#
+# Manage [Contact
+# Routing](https://sensuapp.org/docs/0.26/enterprise/contact-routing.html)
+# configuration with Sensu Enterprise.
+#
+# == Parameters
+#
+# [*ensure*]
+#   String. Whether the check should be present or not
+#   Default: present
+#   Valid values: present, absent
+#
+# [*base_path*]
+#   String.  Where to place the contact JSON configuration file.  Defaults to
+#   `undef` which defers to the behavior of the underlying sensu_contact type.
+#   Default: undef
+#
+# [*config*]
+#   Hash. The configuration data for the contact.  This is an arbitrary hash to
+#   accommodate the various communication channels. For example, `{ "email": {
+#   "to": "support@example.com" } }`.
+#   Default: {}
+define sensu::contact(
+  $ensure = 'present',
+  $base_path = undef,
+  $config = {},
+) {
+  validate_re($ensure, ['^present$', '^absent$'] )
+  validate_hash($config)
+
+  $file_ensure = $ensure ? {
+    'absent' => 'absent',
+    default  => 'file'
+  }
+
+  # handler configuration may contain "secrets"
+  file { "/etc/sensu/conf.d/contacts/${name}.json":
+    ensure => $file_ensure,
+    owner  => 'sensu',
+    group  => 'sensu',
+    mode   => '0440',
+    before => Sensu_contact[$name],
+  }
+
+  sensu_contact { $name:
```

Comment:
instead of a type/provider, what about using sensu_sorted_json() as is done in sensu::write_json? The current layout is to used defined types that then call custom type/providers. What about still calling defined types, but then replacing the type/provider code with sensu_sorted_json()? If possible, I think that would greatly simplify things.